package world_utils.msgs;

import "object.proto";
import "bounding_box.proto";

message WorldUtilsRequest
{
    /* @brief Type of request */
    enum Type{
        /* @brief Spawn entity in simulation */
        SPAWN   = 1;
        /* @brief Move entity in simulation */
        MOVE    = 2;
        /* @brief Remove entity from simulation  */
        REMOVE  = 3;
        /* @brief Start or stop physics simulation */
        PHYSICS = 4;
        /* @brief Pause or resume world */
        PAUSE   = 5;
        /* @brief Request entity or world information */
        STATUS  = 6;
    }

    /* @brief Type of model to be spawned */
    enum ModelType{
        /* @brief Spawn sphere in simulation */
        SPHERE          = 1;
        /* @brief Spawn cylinder in simulation */
        CYLINDER        = 2;
        /* @brief Spawn cube in simulation */
        BOX             = 3;
        /* @brief Spawn custom object from sdf string */
        CUSTOM          = 4;
        /* @brief Spawn custom light from sdf string */
        CUSTOM_LIGHT    = 5;
        /* @brief Spawn a model included in gazebo model path */
        MODEL           = 6;
    }


    /* @brief Type of request */
    required Type                   type            = 1;
    /* @brief Object of request */
    repeated Object                 object          = 2;
    /* @brief World physics state */
    optional bool                   state           = 3;
    /* @brief Bounding box */
    repeated BoundingBox            bounding_box    = 4;
}
